packages/fluence-js/src/internal/compilerSupport/v3impl/callFunction.ts

```diff
@@ -124,9 +138,25 @@ const extractArgs = (
         config = args[numberOfExpectedArgs];
     }
 
+    if (structuredArgs.length !== numberOfExpectedArgs) {
+        throw new Error(`Incorrect number of arguments. Expecting ${numberOfExpectedArgs}`);
+    }
+
+    const argsRes = Object.entries(argumentTypes)
+        .map(([name, _], ix) => ({ [name]: structuredArgs[ix] }))
```

Comment:
There is no need for this "map". Everything can be done inside of the "reduce". You can access index there. Also I prefer the word "index" or just "i" never seen "ix" - it confused me a little bit